packages/webapp/src/components/Task/AddSoilAmendmentProducts/QuantityApplicationRate/index.tsx

```diff
@@ -0,0 +1,249 @@
+    );
+  };
+
+  // Unit and NumberInput accept inline style objects, rather than classnames, to style subcomponents
+  const inputLabelStyles = {
+    color: 'var(--Colors-Neutral-Neutral-500)',
+    fontSize: '16px',
+    lineHeight: 'normal',
+  };
+
+  return (
+    <div className={styles.container}>
+      <Switch
+        leftLabel={formatLabel('ADD_TASK.SOIL_AMENDMENT_VIEW.WEIGHT', isWeight)}
+        label={formatLabel('ADD_TASK.SOIL_AMENDMENT_VIEW.VOLUME', !isWeight)}
+        isToggleVariant
+        onChange={toggleMeasure}
+        checked={!isWeight}
+      />
+      <div className={styles.applicationRateCard}>
+        <div>
+          {/* @ts-ignore */}
+          <Unit
+            label={t('ADD_TASK.SOIL_AMENDMENT_VIEW.QUANTITY')}
+            data-cy="soilAmendment-quantity"
+            register={register}
+            name={TASK_PRODUCT_FIELD_NAMES[isWeight ? 'WEIGHT' : 'VOLUME']}
+            displayUnitName={TASK_PRODUCT_FIELD_NAMES[isWeight ? 'WEIGHT_UNIT' : 'VOLUME_UNIT']}
+            unitType={SOIL_AMENDMENT_UNITS[isWeight ? 'WEIGHT' : 'VOLUME'].units}
+            system={system}
+            hookFormSetValue={setValue}
+            hookFormGetValue={getValues}
+            hookFromWatch={watch}
+            control={control}
+            mode={'onChange'}
+            onChange={onQuantityChange}
+            disabled={isReadOnly}
+            required
+            key={isWeight ? 'weight' : 'volume'}
+            classes={{ label: inputLabelStyles }}
+          />
+          <AreaApplicationSummary
+            locationArea={previewStringValue!}
+            locationAreaUnit={previewStringUnit!}
+            percentOfArea={percent_of_location || 0}
+            locationType={t(`FARM_MAP.MAP_FILTER.${type.toUpperCase()}`).toLocaleLowerCase()}
+          />
+        </div>
+        <div className={clsx(styles.border, styles.advancedSection, isExpanded && styles.expanded)}>
+          <TextButton onClick={toggleExpanded} className={clsx(styles.advancedTitle)}>
+            <span>{t('ADD_TASK.SOIL_AMENDMENT_VIEW.ADVANCED')}</span>
+            <KeyboardArrowDownIcon className={styles.expandIcon} />
+          </TextButton>
+
+          <Collapse id={`application_rate-${productId}`} in={isExpanded} timeout="auto">
+            <div className={styles.sectionBody}>
+              <div className={styles.locationSection}>
+                <NumberInput
+                  name={TASK_PRODUCT_FIELD_NAMES.PERCENT_OF_LOCATION}
+                  control={control}
+                  label={t('ADD_TASK.SOIL_AMENDMENT_VIEW.PERECENT_TO_AMEND')}
+                  min={0}
+                  max={100}
+                  rules={{ required: t('common:REQUIRED') }}
+                  onChange={onPercentLocationChange}
+                  classes={{ label: inputLabelStyles }}
+                  disabled={isReadOnly}
+                  defaultValue={
+                    defaultValues?.[TASK_PRODUCT_FIELD_NAMES.PERCENT_OF_LOCATION] || 100
```

Comment:
Yeah, I agree they would be better synched. Since I need to keep this `100` to populate the default/initial load, I changed the percentOfArea line -- it was only for showing 0% in the "Applied to..." string when the input was cleared, but I think I even like it better this way:

![Screenshot 2024-07-09 at 9 29 23 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/82857964/5825100c-ebc9-466d-93a4-6009cc2ccaea)